### DISCUSSION TITLE
Allow hiding of system boundary for container view

### DISCUSSION THREAD

**Discussion by ncasaux** [structurizr/java#216] [Q&A]
### Description

Hello @simonbrowndotje ,

Would you consider adding a boolean property named something like `softwareSystemBoundaryVisible` on a container view, to hide the rectangle with is currently automatically added in the generated PlantUML code ?
The behaviour would be very similar to the existing property `enterpriseBoundaryVisible` available on system landscape view.

The reason behind this is that with a system which have lots of containers, external systems and relationships, the rectange around the system makes GraphViz generate unpleasant diagram (as shown in attached file, the relationships gets all sticked together because of the alignement inside the rectangle), which are actually much better without this rectangle.

![image](https://github.com/structurizr/java/assets/16475875/940a297a-fb23-4f82-8deb-8846ef003936)

Thanks in advance!

### Priority

Low

### Resolution

I'm willing to add this feature myself and raise a PR

**Comment by simonbrowndotje** [comment]
This is probably better done via a view property in the same way `structurizr.softwareSystemBoundaries` is supported for component views (see https://structurizr.com/help/properties). But I wouldn't recommend removing the software system boundaries because, unless you also modify how the containers are rendered, the diagrams become less clear. For example, consider this DSL:

```
workspace {

    model {
        a = softwareSystem "A" {
            aa = container "API"
        }
        b = softwareSystem "B" {
            bb = container "API"
        }

        aa -> bb "Requests data from"
    }

    views {
        container a {
            include aa bb
            autoLayout lr
        }
    }
    
}
```

And here's the diagram that would be generated if you remove the software system boundaries.

![image](https://github.com/structurizr/java/assets/1009874/7d48c73c-c751-4637-b325-ab9f40efb544)

**Comment by ncasaux** [comment]
Thank you @simonbrowndotje for your reply.

In my context, I'm not using Structurizr UI, but only the Structurizr CLI with the PlantUML Exporter. That's why adding a property on the container view seemed me as the most suitable option.

Your recommandation about removing the software system boundaries is perfectly valid, and I agree with it. However, in my context, I do never display containers of different systems on the same view, so there can't really have a confusion.

With system boundary:
![image](https://github.com/structurizr/java/assets/16475875/551ee6dd-49d4-4a0a-957a-0e815e77089b)

Without system boundary:
![image](https://github.com/structurizr/java/assets/16475875/a035b7da-7e9b-4090-aba4-909efd2502db)

The differences are not huge on this example because there are not that much containers/relationships, but it adds clarity for more complex/intricated diagrams.

**Reply by simonbrowndotje** [reply]
I don't like the potential ambiguity, so it's not a feature that I want to add - you're welcome to fork the PlantUML exporter for your own use case though.

**Reply by ncasaux** [reply]
No problem, I can understand. We are already using a forked version of the exporter with this feature.
Thanks for your feedback.
